File: frontend/validate.js
const fs = require('fs');
const path = require('path');

// Simple validation script to check frontend structure
const checks = [
  { name: 'Auth context', path: 'src/context/AuthContext.jsx' },
  { name: 'Auth library', path: 'src/lib/auth.js' },
  { name: 'Middleware', path: 'src/middleware.js' },
  { name: 'Root layout', path: 'src/app/layout.jsx' },
  { name: 'Home page', path: 'src/app/page.jsx' },
  { name: 'Login page', path: 'src/app/login/page.jsx' },
  { name: 'Register page', path: 'src/app/register/page.jsx' },
  { name: 'Profile page', path: 'src/app/profile/[id]/page.jsx' },
  { name: 'Global styles', path: 'src/styles/globals.css' },
  { name: 'Package config', path: 'package.json' },
  { name: 'Next config', path: 'next.config.js' },
  { name: 'Tailwind config', path: 'tailwind.config.js' },
];

console.log('🔍 Validating Frontend Structure...\n');

let allValid = true;
checks.forEach(check => {
  const fullPath = path.join(__dirname, check.path);
  if (fs.existsSync(fullPath)) {
    console.log(`✅ ${check.name}: ${check.path}`);
  } else {
    console.log(`❌ ${check.name}: ${check.path} (NOT FOUND)`);
    allValid = false;
  }
});

console.log();
if (allValid) {
  console.log('✅ All files present!\n');
  console.log('📝 Next Steps:');
  console.log('1. Run: npm install');
  console.log('2. Run: npm run dev');
  console.log('3. Open: http://localhost:3000');
  console.log('4. Test registration and login flows');
} else {
  console.log('❌ Some files are missing!');
  process.exit(1);
}

File: backend/validate.go
package main

import (
	"fmt"
	"os"
)

// Simple validation script to check code structure
func main() {
	checks := []struct {
		name string
		path string
	}{
		{"Database initialization", "pkg/db/sqlite/sqlite.go"},
		{"User model", "models/user.go"},
		{"Auth handlers", "pkg/handlers/auth.go"},
		{"User handlers", "pkg/handlers/user.go"},
		{"Auth middleware", "queries/middleware/auth.go"},
		{"CORS middleware", "queries/middleware/cors.go"},
		{"Server setup", "server/server.go"},
		{"Main entry", "cmd/app/main.go"},
		{"Users table migration up", "pkg/db/migration/sqlite/000001_create_users_table.up.sql"},
		{"Users table migration down", "pkg/db/migration/sqlite/000001_create_users_table.down.sql"},
	}

	fmt.Println("🔍 Validating Backend Structure...")
	fmt.Println()

	allValid := true
	for _, check := range checks {
		if _, err := os.Stat(check.path); err == nil {
			fmt.Printf("✅ %s: %s\n", check.name, check.path)
		} else {
			fmt.Printf("❌ %s: %s (NOT FOUND)\n", check.name, check.path)
			allValid = false
		}
	}

	fmt.Println()
	if allValid {
		fmt.Println("✅ All files present!")
		fmt.Println()
		fmt.Println("📝 Next Steps:")
		fmt.Println("1. Install GCC/MinGW-w64 for SQLite support")
		fmt.Println("2. Run: $env:CGO_ENABLED=1; go build ./cmd/app")
		fmt.Println("3. Run: go test -v ./models/...")
		fmt.Println("4. Run: ./app.exe")
		fmt.Println("5. Test the API endpoints")
	} else {
		fmt.Println("❌ Some files are missing!")
		os.Exit(1)
	}
}
